packages/cli-kit/src/cli.ts

```diff
@@ -11,3 +13,24 @@ export const globalFlags = {
     env: 'SHOPIFY_FLAG_VERBOSE',
   }),
 }
+
+export async function isCliProject(directory: string) {
```

Comment:
I'd consider defining an internal plugin contract around this and get the `@shopify/app`, `@shopify/cli-hydrogen`, and `@shopify/theme` packages to implement it. Otherwise `@shopify/cli-kit` has knowledge on the packages that are built on it, and it shouldn't.  https://en.wikipedia.org/wiki/Dependency_inversion_principle

![Untitled Diagram-Page-3 drawio](https://user-images.githubusercontent.com/663605/191442757-bb1e8d1f-16a5-4519-978a-3eb7071495d3.png)

Reply:
Finally I have reused the existing method `localCliPackage` to detect the using of a local CLI instance